app/src/main/java/org/wikipedia/feed/onthisday/OnThisDayViewModel.kt

```diff
@@ -35,14 +39,20 @@ class OnThisDayViewModel(bundle: Bundle) : ViewModel() {
         _uiState.value = Resource.Loading()
         viewModelScope.launch(handler) {
             val response = ServiceFactory.getRest(wikiSite).getOnThisDay(calendar[Calendar.MONTH] + 1, calendar[Calendar.DATE])
+            val hasParentLanguageCode = !WikipediaApp.instance.languageState.getDefaultLanguageCode(wikiSite.languageCode).isNullOrEmpty()
+            if (hasParentLanguageCode) {
+                val events = response.allEvents().filter { it.pages.isNotEmpty() }
+                val eventPages = events.map { event ->
+                    async { L10nUtil.getPagesForLanguageVariant(event.pages, wikiSite) }
```

Comment:
I'm not sure this is a good idea. This will make an API call for each page in each event, which could be **hundreds** of API calls.
This is really a case that should probably be fixed on the server side of the `/feed/onthiday` API, not patched over on the client side.

Reply:
It makes sense to me. 😢